docs/self-hosted-getting-started.md

```diff
@@ -0,0 +1,99 @@
+          POSTGRES_PASSWORD: postgres
+          OPENAI_API_KEY: your-api-key
+        ports:
+          - "5432:5432"
+        volumes:
+          - ./data:/var/lib/postgresql/data
+    
+      vectorizer:
+        image: timescale/pgai-vectorizer-worker:rc
+        environment:
+          VECTORIZER_DB_URL: postgres://postgres:postgres@db:5432/postgres
+          OPENAI_API_KEY: your-api-key
+        depends_on:
+          - db
+    ```
+
+2. Connect to your db instance with your DB client of choice. We need to enable the pgai extension and create a simple blog table with the following schema:
+    ```sql
+    CREATE EXTENSION IF NOT EXISTS ai CASCADE;
+    CREATE TABLE blog (
+        id SERIAL PRIMARY KEY,
+        title TEXT,
+        authors TEXT,
+        contents TEXT,
+        metadata JSONB
+    );
+    ```
+   
+3. Insert some data into your new table. E.g. you can run these insert statements:
+    ```sql
+    INSERT INTO blog (title, authors, contents, metadata)
+    VALUES
+    ('Getting Started with PostgreSQL', 'John Doe', 'PostgreSQL is a powerful, open source object-relational database system...', '{"tags": ["database", "postgresql", "beginner"], "read_time": 5, "published_date": "2024-03-15"}'),
+
+    ('10 Tips for Effective Blogging', 'Jane Smith, Mike Johnson', 'Blogging can be a great way to share your thoughts and expertise...', '{"tags": ["blogging", "writing", "tips"], "read_time": 8, "published_date": "2024-03-20"}'),
+
+    ('The Future of Artificial Intelligence', 'Dr. Alan Turing', 'As we look towards the future, artificial intelligence continues to evolve...', '{"tags": ["AI", "technology", "future"], "read_time": 12, "published_date": "2024-04-01"}'),
+
+    ('Healthy Eating Habits for Busy Professionals', 'Samantha Lee', 'Maintaining a healthy diet can be challenging for busy professionals...', '{"tags": ["health", "nutrition", "lifestyle"], "read_time": 6, "published_date": "2024-04-05"}'),
+
+    ('Introduction to Cloud Computing', 'Chris Anderson', 'Cloud computing has revolutionized the way businesses operate...', '{"tags": ["cloud", "technology", "business"], "read_time": 10, "published_date": "2024-04-10"}'); 
+    ```
+
+4. Now we need to create a vectorizer for the table. The vectorizer worker will pick this one up and create the corresponding embeddings for us:
+    ```sql
+    SELECT ai.create_vectorizer(
+       'blog'::regclass,
+       destination => 'blog_contents_embeddings',
+       embedding => ai.embedding_openai('text-embedding-3-small', 768),
+       chunking => ai.chunking_recursive_character_text_splitter('contents'),
+       scheduling => ai.scheduling_none()
+    );
+    ```
+    If you check the logs of the vectorizer worker, you should see that it has picked up the table and is processing it.
+
+
+5. You'll need to restart the vectorizer now, since it fails to run if pgai is not installed or no vectorizer is defined yet. A simple `docker-compose up -d` should do the trick.
```

Comment:
This is quite ugly, can we change the CLI so that it works even if pgai is not installed and there is no vectorizer yet? logging a warning/error is fine but the container crashing is at least for this tutorial annoying (I think also in normal use?)

Reply:
This can be solve by starting first the db with `docker compose up -d db`, and at this point we run `docker compose up -d pgai-vectorizer-worker`.

Reply:
If we go this route we need to tell them that if they want to see the logs use

```
docker compose logs -f
```

Just in case something goes wrong they can have the logs.

Reply:
I like the split up `docker-compose up` so far, makes it simple and removes the need for depends_on etc. 
For logs I added this at the bottom in the troubleshooting section.